response.go

```diff
@@ -70,19 +81,19 @@ func (err connectionError) Timeout() bool {
 	return true
 }
 
-// serverError represents internal server errors.
+// ServerError represents internal server errors.
 // Implements `net.Error` interface.
-type serverError string
+type ServerError string
```

Comment:
My idea was, use this error as `net.Error` error in case if you want to retry. So you can always do follow things:
```
result, err := someReallyDeepNestedCall()
if err != nil {
  switch v := err.(type) {
  case net.Error:
  case *otherpackage.Error:
  }
}
```

Or you can assert to your own interface, for example:
```go
type temporary interface {
    Temporary() bool
}

type timeout interface {
     Timeout() bool
}
```
and  `serverError` will implement both interfaces. So I am curious about your case, how you want yo use it. Maybe it make sense just check for error itself (`if err == fcm.ErrInternalServerError`).